tests/test_preprocess_artifacts.py

```diff
@@ -39,10 +58,16 @@ def check_step_artifact_present(self) -> None:
     def check_step_artifact_absent(self) -> None:
         assert not os.path.exists(self.artifact_path)
 
-    def create_artifact_file(self, directory: pathlib.Path) -> None:
-        precreated_artifact: pathlib.Path = directory / self.artifact_name
-        with open(precreated_artifact, "w", encoding="utf-8") as f:
-            f.write("pre-created artifact content")
+    def check_step_dir_artifact_present(self) -> None:
+        assert os.path.exists(self.dir_archive)
+        with zipfile.ZipFile(self.dir_archive) as dir_zip:
```

Comment:
This check assumes '--no-archive' option is absent by default.
This could change in the future.
I suggest to explicitly set it in settings to False or to implement tests for both cases.